blackbox/hdfs/src/tests.py

```diff
@@ -100,26 +102,48 @@ def _get_hadoop(self):
                             'hadoop-{version}'.format(version=HADOOP_VERSION))
 
     def setUp(self):
-        cmd = ['bash',
-               self.hadoop_bin, 'jar',
-               self.hadoop_mapreduce_client, 'minicluster',
-               '-nnport', NN_PORT, '-nomr', '-D', 'dfs.replication=0']
+        cmd = [
+            self.hadoop_bin, 'jar',
+            self.hadoop_mapreduce_client, 'minicluster',
+            '-nnport', NN_PORT, '-nomr', '-format',
+            '-D', 'dfs.replication=0',
+            '-D', 'dfs.client.use.datanode.hostname=true',
+            '-D', 'dfs.datanode.use.datanode.hostname=true',
+        ]
 
         JAVA_HOME = os.environ.get('JAVA_HOME', '/usr/lib/jvm/java-8-openjdk/')
+
+        print('')
+        print('>>> ' + 'JAVA_HOME=' + JAVA_HOME)
+        print('>>> ' + 'HADOOP_CLASSPATH=' + self.yarn_server_jar)
+        print('>>> ' + ' '.join(cmd))
+        print('')
+
         self.p = subprocess.Popen(
             cmd,
             cwd=os.path.dirname(self.hadoop_bin),
             env={
                 'JAVA_HOME': JAVA_HOME,
                 'HADOOP_CLASSPATH': self.yarn_server_jar
-            })
-        elapsed = 0.0
-        while not is_up('localhost', int(NN_PORT)) and elapsed < 5:
-            time.sleep(0.1)
-            elapsed += 0.1
+            },
+            stdout=subprocess.PIPE,
```

Comment:
Could this cause the PIPE to get full and the process get stuck if there is a lot of output?